grails-app/services/org/pih/warehouse/inventory/ProductAvailabilityService.groovy

```diff
@@ -453,18 +454,34 @@ class ProductAvailabilityService {
         return quantityMap
     }
 
-    Map<Product, Map<Location, Integer>> getQuantityOnHandByProduct(Location[] locations) {
+    Map<Product, Map<Location, Integer>> getQuantityOnHandByProduct(Location[] locations, String[] categories, Boolean includeCategoryChildren) {
+        // Mysql throws an error when query with "category.id in (empty set)" is executed, but it works when NULL is included in this set
+        String queryWithCategories = categories ? "and category.id in (:categories)" : 'and category.id in (NULL)'
+
+        // Getting all subcategories of picked categories
+        def categoriesWithChildren = categories.collect { String categoryId ->
+            Category category = Category.load(categoryId)
+            category.getChildren() + category
+        }.flatten().collect { it.id }
+
+        // I am checking if "include all products in all subcategories" checkbox is checked to add children categories
+        def availableCategories = includeCategoryChildren ? categoriesWithChildren : categories
```

Comment:
if so, I'd do something to avoid calculating (fetching, collecting, etc) `categoriesWithChildren` from above, if you might not even use it if the `includeCategoryChildren` is not checked.